confgenerator/config.go

```diff
@@ -793,83 +795,46 @@ var (
 	}
 )
 
-// mapKeys returns keys from a map[string]Any as a map[string]bool.
-func mapKeys(m interface{}) map[string]bool {
-	keys := map[string]bool{}
-	switch m := m.(type) {
-	case loggingReceiverMap:
-		for k := range m {
-			keys[k] = true
-		}
-	case map[string]LoggingProcessor:
-		for k := range m {
-			keys[k] = true
-		}
-	case map[string]*LoggingPipeline:
-		for k := range m {
-			keys[k] = true
-		}
-	case metricsReceiverMap:
-		for k := range m {
-			keys[k] = true
-		}
-	case metricsProcessorMap:
-		for k := range m {
-			keys[k] = true
-		}
-	case map[string]*MetricsPipeline:
-		for k := range m {
-			keys[k] = true
-		}
-	default:
-		panic(fmt.Sprintf("Unknown type: %T", m))
+// sortedKeys returns sorted keys from a Set if the Set has a type that can be ordered.
+func sortedKeys[K constraints.Ordered, V any](m map[K]V) []K {
+	keys := []K{}
+	for k := range m {
+		keys = append(keys, k)
 	}
+	sort.Slice(keys, func(i, j int) bool {
+		return keys[i] < keys[j]
+	})
 	return keys
 }
 
-// sortedKeys returns keys from a map[string]Any as a sorted string slice.
-func sortedKeys(m interface{}) []string {
-	var r []string
-	for k := range mapKeys(m) {
-		r = append(r, k)
-	}
-	sort.Strings(r)
-	return r
-}
-
 // findInvalid returns all strings from a slice that are not in allowed.
```

Comment:
This only returns 1 invalid string, not all. The comment might need to be updated

Reply:
While trying to write a better descriptive comment I realized that this function is pretty unnecessary and decided to inline it.